File: templates/TwoSum.java
import java.util.*;

public class temp {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (seen.containsKey(complement)) {
                return new int[]{seen.get(complement), i};
            }
            seen.put(nums[i], i);
        }
        return new int[]{};
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt(); // Read size of array
        int[] nums = new int[n];
        for (int i = 0; i < n; i++) {
            nums[i] = scanner.nextInt(); // Read each element
        }
        int target = scanner.nextInt(); // Read target

        temp solution = new temp();
        int[] result = solution.twoSum(nums, target);

        if (result.length == 2) {
            System.out.println("[" + result[0] + ", " + result[1] + "]");
        } else {
            System.out.println("No solution found");
        }

        scanner.close();
    }
}

File: templates/TwoSum.cpp
#include <vector>
#include <iostream>
using namespace std;

// Helper function to parse input into a vector
vector<int> parseInput() {
    int n;
    cin >> n; // Read the size of the array
    vector<int> nums(n);
    for (int i = 0; i < n; i++) {
        cin >> nums[i]; // Read each number
    }
    int target;
    cin >> target; // Read the target
    return nums;
}

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // Your code here
        // Implement the logic to find two indices whose elements sum to target
        // Return a vector containing the two indices
        return {};
    }
};

int main() {
    // Read input
    vector<int> nums = parseInput();
    int target;
    cin >> target; // Read target (already handled by parseInput in this case)

    // Create solution instance and get result
    Solution solution;
    vector<int> result = solution.twoSum(nums, target);

    // Output result
    if (result.size() == 2) {
        cout << result[0] << " " << result[1] << endl;
    } else {
        cout << "No solution found" << endl;
    }

    return 0;
}